### PR TITLE
2.x: Feature/#4876 more null checks

### PR DESCRIPTION
**Pull request by jschneider**
I have added some more null checks to help the static code analysis.
I think I have also fixed one possible NPE in NewThreadWorker

### CODE REVIEW

File: src/main/java/io/reactivex/internal/operators/flowable/FlowableCombineLatest.java [hunk 1]
```diff
@@ -73,6 +76,7 @@ public void subscribeActual(Subscriber<? super R> s) {
         Publisher<? extends T>[] a = array;
         int n;
         if (a == null) {
+            assert iterable != null; //either array or iterable are initialized with non null values
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/flowable/FlowableCombineLatest.java]: Asserts are off by default but they increase the bytecode size; besides we don't use `assert` it anyway.

File: src/main/java/io/reactivex/internal/operators/flowable/FlowableWindow.java [hunk 2]
```diff
@@ -246,6 +246,7 @@ public void onNext(T t) {
 
             if (i == size) {
                 window = null;
+                assert w != null;
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/flowable/FlowableWindow.java]: Please remove these asserts.

File: src/main/java/io/reactivex/internal/operators/flowable/FlowableWithLatestFromMany.java [hunk 3]
```diff
@@ -33,21 +35,22 @@
...
-    public FlowableWithLatestFromMany(Publisher<T> source, Publisher<?>[] otherArray, Function<? super Object[], R> combiner) {
+    public FlowableWithLatestFromMany(@NonNull Publisher<T> source, @NonNull Publisher<?>[] otherArray, Function<? super Object[], R> combiner) {
         super(source);
         this.otherArray = otherArray;
         this.otherIterable = null;
         this.combiner = combiner;
     }
 
-    public FlowableWithLatestFromMany(Publisher<T> source, Iterable<? extends Publisher<?>> otherIterable, Function<? super Object[], R> combiner) {
+    public FlowableWithLatestFromMany(@NonNull Publisher<T> source, @NonNull Iterable<? extends Publisher<?>> otherIterable, Function<? super Object[], R> combiner) {
```
Comment (akarnokd) [inline src/main/java/io/reactivex/internal/operators/flowable/FlowableWithLatestFromMany.java]: `combiner` needs annotation

File: src/test/java/io/reactivex/schedulers/NewThreadSchedulerTest.java [hunk 4]
```diff
@@ -115,4 +115,18 @@ public void run() {
 
         assertEquals(0, calls[0]);
     }
+
+    @Test
+    public void npe() throws Exception {
```
Comment (akarnokd) [inline src/test/java/io/reactivex/schedulers/NewThreadSchedulerTest.java]: This could be named a bit more informatively.

**Review by akarnokd — CHANGES_REQUESTED** [review]

File: src/main/java/io/reactivex/internal/operators/flowable/FlowableCombineLatest.java (hunk 1 above)
Comment (jschneider) [reply src/main/java/io/reactivex/internal/operators/flowable/FlowableCombineLatest.java]: ok. removed it.
Do you have a standard way to suppress false positives?

File: src/main/java/io/reactivex/internal/operators/flowable/FlowableWindow.java (hunk 2 above)
Comment (jschneider) [reply src/main/java/io/reactivex/internal/operators/flowable/FlowableWindow.java]: done

File: src/main/java/io/reactivex/internal/operators/flowable/FlowableWithLatestFromMany.java (hunk 3 above)
Comment (jschneider) [reply src/main/java/io/reactivex/internal/operators/flowable/FlowableWithLatestFromMany.java]: done

File: src/test/java/io/reactivex/schedulers/NewThreadSchedulerTest.java (hunk 4 above)
Comment (jschneider) [reply src/test/java/io/reactivex/schedulers/NewThreadSchedulerTest.java]: yep. added some comments

**Review by akarnokd — APPROVED** [review]

**Pull request merged**